tasks/leiningen/project_version.clj

```diff
@@ -0,0 +1,20 @@
+(ns leiningen.project-version
+  (:require [clojure.java.shell :as sh]))
+
+(defn project-version [project & args]
+  (println (:version project)))
+
+(defn- version-from-git []
+  (let [{:keys [exit out]} (sh/sh "git" "describe" "--tags" "--dirty" "--long")]
+    (assert (= 0 exit))
+    ;;"19.04-1.0.2-alpha-22-g9452e91b-dirty"
+    (let [[_ tag ahead sha dirty] (re-find #"(.*)\-(\d+)\-([0-9a-z]*)(\-dirty)?$" (.trim out))
```

Comment:
Have a strong feeling this will be a reflection warning for `.trim` here? Can hint out as `^String` or use `clojure.string/trim` to indirectly achieve the same thing.